pgdb/v1/fts.go

```diff
@@ -112,9 +112,109 @@ func lemmatizeDocs(docs []*SearchContent, additionalFilters ...jargon.Filter) []
 	return rv
 }
 
-func camelSplitDoc(docValue string, doc *SearchContent) []lexeme {
+// snakeSubTokensSplitDoc tokenizes foo bar baz in foo_bar_baz.
+func snakeSubTokensSplitDoc(docValue string, wordBuffer bytes.Buffer, doc *SearchContent) []lexeme {
+	wordBuffer.Reset()
+	rv := make([]lexeme, 0, 8)
+	var pos = 1
+	var prev rune
+	for _, r := range docValue {
+		if prev == 0 {
+			prev = r
+			continue
+		}
+		if prev == '_' {
+			if wordBuffer.Len() == 0 { // no current word
+				if unicode.IsPunct(r) || unicode.IsSpace(r) || unicode.IsControl(r) || unicode.IsSymbol(r) {
+					prev = r
+					continue
+				}
+				// starting a new word
+				if _, e := wordBuffer.WriteRune(r); e != nil {
+					wordBuffer.Reset()
+					continue
+				}
+			}
+		} else if wordBuffer.Len() > 0 {
+			// in a current word, do we append or end?
+			switch {
+			case r == '_' || unicode.IsPunct(r) || unicode.IsSpace(r) || unicode.IsControl(r) || unicode.IsSymbol(r):
+				if utf8.RuneCount(wordBuffer.Bytes()) >= minWordSize {
+					// have a word, current is a rune that doesn't continue the word so end current word
+					rv = append(rv, lexeme{strings.ToLower(wordBuffer.String()), pos, doc.Weight})
+				}
+				wordBuffer.Reset()
+			default:
+				// in word and current rune continues word so continue appending
+				if _, e := wordBuffer.WriteRune(r); e != nil {
+					wordBuffer.Reset()
+					continue
+				}
+			}
+		}
+		prev = r
+		pos += 1
+	}
+	if utf8.RuneCount(wordBuffer.Bytes()) >= minWordSize {
+		rv = append(rv, lexeme{strings.ToLower(wordBuffer.String()), pos, doc.Weight})
+	}
+	return rv
+}
+
+// snakeFullTokensSplitDoc tokenizes foo_bar_baz from foo_bar_baz.
```

Comment:
Comment out of date? Seems like it would get `[]string{foo, bar, baz}` from `foo_bar_baz`